src/cargo/util/command_prelude.rs

```diff
@@ -504,6 +512,29 @@ pub trait ArgMatchesExt {
         build_config.build_plan = self.is_valid_and_present("build-plan");
         build_config.unit_graph = self.is_valid_and_present("unit-graph");
         build_config.future_incompat_report = self.is_valid_and_present("future-incompat-report");
+
+        if self.is_valid_and_present("timings") {
+            for timing_output in self._values_of("timings") {
+                for timing_output in timing_output.split(',') {
+                    let timing_output = timing_output.to_ascii_lowercase();
+                    let timing_output = match timing_output.as_str() {
+                        "html" => TimingOutput::Html,
+                        "json" => {
+                            config
+                                .cli_unstable()
+                                .fail_if_stable_opt("--timings=json", 7405)?;
```

Comment:
For now could we make it unstable to pass anything to `--timings` at all? Make even `--timings=html` unstable? (only `--timings` would be stable)

Reply:
@alexcrichton Interesting! I could do that, sure. What's the motivation?

Reply:
I could imagine doing the reverse, making `--timings=html` stable but `--timings` not, which would allow us to easily change the defaults in the future and not nail them down right now.

What capability do we gain if `--timings=html` is unstable? (Except perhaps the ability to rename `html`?)

Reply:
Mostly just being conservative. I sort of doubt that anyone needs anything other than `--timings` for now so I'm not surer why we would also want to offer `--timings=html` initially

Reply:
Alright, done.